### DISCUSSION TITLE
using memfs with webpack

### DISCUSSION THREAD

**Discussion by arunkumar413** [webpack/webpack#16517] [General]
Hi Team,

I'd like to use a virtual file system for input and output options. Could you please share more info on how to use memfs with webpack.

I referred to this doc ( https://webpack.js.org/api/node/#custom-file-systems). It uses `require` statements but I'd like to use import statements.

Is the below code snippet the correct approach to use `memfs` with webpack? If not please share the correct code.

```
import { fs, vol } from 'memfs';

 let compiler = webpack({}, function (err, stats) {
    if (err || stats.hasErrors()) {
      // handle error
    }
  });
  compiler.inputFileSystem = fs;
  compiler.outputFileSystem = fs;
```

**Comment by alexander-akait** [comment]
You can use `import` (just set `type: "module"` in `package.json` or use the `mjs` extension), the code is right
